src/toolkit/mat_query.cc

```diff
@@ -41,6 +41,19 @@ double MatQuery::atom_frac(Nuc nuc) {
   return v[nuc];
 }
 
+double MatQuery::atom_frac(std::set<Nuc> nucs) {
+  CompMap v = m_->comp()->atom();
+  compmath::Normalize(&v);
+  
+  double frac_tot = 0;
+  std::set<Nuc>::iterator it ;
+  for (it = nucs.begin(); it != nucs.end(); ++it) {
+    frac_tot += v[*it];
```

Comment:
You should check if `*it` is in `v` first, so that extra nuclides aren't added to the map.